### DISCUSSION TITLE
Intermittent failure of the Permanently Detach feature

### DISCUSSION THREAD

**Discussion by tmccoid-tech** [tmccoid-tech/extract-em#24] [Announcements]
As of Extract 'Em! version 1.2.1 and Thunderbird 128, users of the extension were provided with an option to delete extracted attachments from the source email message in a separate operation following extraction.  A number of users have reported that this process fails, and the extension stalls.

Thunderbird extensions communicate with the client application through an API; in this case, the process fails on the first attempt at calling the API method/function _deleteAttachments()_; no response is returned after this method is called, and no error is thrown (i.e., no indication that an error occurred is provided.)

The problem is not universal; the developer has never been experienced the issue having run dozens of tests.  It appears, for any given user, to be all or nothing -- either all the attachments in question are successfully deleted (excepting properly reported/handled errors) and the process completes, or none of them are and the process stalls.

This leads the developer to believe that the problem has to do with an "environment" problem -- something about certain users' systems is conflicting with the method call, such as a Thunderbird setting, an OS setting, an anti-virus program, or even another extension. 

If you have experienced this issue, if you could post a comment in this discussion and provide the following information at a minimum:

1) Email account type (pop, imap, etc,.)

the Extract 'Em! developer would be appreciative.

UPDATE: The Thunderbird API team believes the issue may be specific to IMAP accounts.  This is the key piece of information required at this time.  Thank you.

**Comment by B-M-B** [comment]
Thanks for the Add-on, extracting works fine with me, but deletion doesn't.
I am running Thunderbird on following system:

Operating System: EndeavourOS 
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.6.0
Qt Version: 6.7.3
Kernel Version: 6.11.1-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 8 × Intel® Core™ i5-8250U CPU @ 1.60GHz
Memory: 7,5 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics 620
Manufacturer: LENOVO
Product Name: 81C3
System Version: Lenovo YOGA 720-13IKB

The process stops without error in following window

![extract-em no progress detach window](https://github.com/user-attachments/assets/d1e09654-aada-4d09-8cfe-a3be69d874be)

**Reply by tmccoid-tech** [reply]
Thank you for the detailed report.

**Reply by tmccoid-tech** [reply]
Would the account against which you are attempting to perform the Permanently Detach operation be an IMAP account?  The problem may be related to this particular account type.  Thanks.

**Reply by B-M-B** [reply]
Yes, it is an IMAP account

**Reply by B-M-B** [reply]
Is it generally not possible to detach attachments when using IMAP accounts? Or is it possible to implement this?

**Reply by tmccoid-tech** [reply]
It is currently not possible due to a defect in the Thunderbird API (which is beyond my control).  The API team believes they have a handle on the problem and with any luck a fix will be issued with a forthcoming release of Thunderbird.

I will update this discussion when that fix has been made generally available.

**Reply by B-M-B** [reply]
Thanks for your feedback

**Comment by tmccoid-tech** [comment]
I have some good news and some bad news regarding this defect.

It is now all but certain that the problem is restricted to IMAP accounts.  I was able to reproduce the error with an IMAP account.

Furthermore, it is very likely that the root cause has been discovered and a fix manufactured by the Thunderbird API team.

However, this fix may not make it into the current Thunderbird ESR (extended support release), which is the standard version for most users.  It is planned for Beta version 133 which is expected to be released at the end of October.  If it is introduced into the ESR, it will not be generally available until late November at the earliest; otherwise it will not be available until the next ESR which I don't believe will be released until the middle of 2025.

Once 133 is released, I am going to test myself to see if the patch does indeed resolve the problem, and if so I may ask if some of the more intrepid amongst you are willing to test with that version.

Thanks for your patience.

**Comment by tmccoid-tech** [comment]
A belated update for anyone who is still paying attention to this discussion: the attempted fix for the IMAP detachment problem did not in fact remedy it.  I will continue to work with the Thunderbird API team to get this issue resolved.

**Comment by tmccoid-tech** [comment]
There are some further developments to report.

The problem no longer occurs as of Thunderbird 136 (a beta version not generally available); unfortunately the patch which fixes the problem will not be back-ported to Thunderbird 128 (the current ESR version).

However, a workaround appears to have been discovered which can be used when Extract 'Em! 1.6.5 is released -- probably in a day or two of this comment.

As strange as it sounds, if, prior to clicking Proceed (with the detachment operation), a message containing an attachment slated for detachment is selected in the Thunderbird message pane, the detachment operation succeeds.  I have run a number of tests using this approach and have not experienced a single stall/hang.

The problem may not have been universal to IMAP accounts of all providers; it certainly occurred with Gmail (and one other outfit), but that is academic.

In any event, I am afraid that is the best I can do.  If the workaround fails for you, you can either install the latest beta or sit tight for the next Thunderbird ESR.

**Comment by chrestir** [comment]
Hi, 
excellent add-on, just what I needed!
Having problem with attachments sort of remaining after deletion though. 
 Selecting a message prior to Proceed did not do the trick. 
The icon indicating an attachment present remains but the attachment itself seems to be removed. See screenshots below.
The folder I'm working on is in the Local Folders if that matters.

Thunderbird v.140.4.0esr (64-bit)
Extract'Em v.1.888.

<img width="132" height="144" alt="2025-10-31 13-38" src="https://github.com/user-attachments/assets/5d7b4ae2-68a6-4bec-a969-7290029965ca" />

<img width="751" height="205" alt="2025-10-31 13-32_1" src="https://github.com/user-attachments/assets/a8997bbf-b006-4154-9743-52c013beb71a" />


/chrestir

**Reply by tmccoid-tech** [reply]
This is the default behavior of Thunderbird.  If you manually delete an attachment in the Thunderbird UI you will find that the icon also persists following deletion.

I have no means of changing this; it would need to be addressed by the Thunderbird development team.
